src/services/redisSessionStorage.ts

```diff
@@ -48,19 +50,29 @@ export default class RedisSessionStorage implements ISessionStorage {
       );
     });
 
-    const [setSessionTokenResult, setWalletTokenResult] = await Promise.all([
+    const newSessionInfo: SessionInfo = {
+      sessionToken: user.session_token,
+      timestamp: new Date()
+    };
+    const setUserTokenInfo = new Promise<Either<Error, boolean>>(resolve => {
+      this.redisClient.set(
+        `${userKeyPrefix}${user.fiscal_code}-${sessionKeyPrefix}${
+          user.session_token
+        }`,
+        JSON.stringify(newSessionInfo),
+        "EX",
+        this.tokenDurationSecs,
+        (err, response) => resolve(this.singleStringReply(err, response))
+      );
+    });
+    const setPromisesResult = await Promise.all([
       setSessionToken,
-      setWalletToken
+      setWalletToken,
+      setUserTokenInfo
     ]);
-
-    if (isLeft(setSessionTokenResult) || isLeft(setWalletTokenResult)) {
-      return left<Error, boolean>(new Error("Error setting the token"));
-    }
-
-    if (!setSessionTokenResult.value || !setWalletTokenResult.value) {
+    if (setPromisesResult.some(_ => isLeft(_) || !_.value)) {
       return left<Error, boolean>(new Error("Error setting the token"));
```

Comment:
the error should be a little more verbose now, as it's not just setting a token, perhaps `Error while storing the user session` - perhaps we can also log the errors with something like:

```ts
setPromisesResult.filter(isLeft).map(_ => _.value.message).join("|");
```

Reply:
fixed on 2354a0b